python/ray/util/client/dataclient.py

```diff
@@ -66,9 +70,16 @@ def _data_main(self) -> None:
                     # This is not being waited for.
                     logger.debug(f"Got unawaited response {response}")
                     continue
-                with self.cv:
-                    self.ready_data[response.req_id] = response
-                    self.cv.notify_all()
+                if response.req_id in self.asyncio_waiting_data:
+                    cb = self.asyncio_waiting_data.pop(response.req_id)
+                    try:
+                        cb(response)
+                    except Exception:
+                        logger.exception("Callback error:")
+                else:
```

Comment:
a regular ref can have both `_on_completed` and `ray.get`. I don't think we need the else here.

Reply:
Those would be under different request IDs. `ready_data` is used for signaling to `blocking send`.